### PR TITLE
Added course schedule functionalities

### PR DESCRIPTION
**Pull request by Daniel-olaO**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -50,15 +50,11 @@ function App() {
             element={
               <Students />
             } />
-        </Route>
-        <Route
-          path='/student/:id'
-          element={
-            <ProtectedRoute isAuth={isLoggedIn}>
```
Comment (Russell-Pollari) [inline src/App.js:57]: why remove this?

File: src/components/CourseCatalog.jsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        .then((data) => {
+          setCourses(data);
+          setLoading(false);
+        },
+        )
+        .catch((err)=>console.log(err));
+  }, []);
+  if (loading) {
+    return <Loading />;
+  } else {
```
Comment (Russell-Pollari) [inline src/components/CourseCatalog.jsx:53]: with an early return in the condition above, you don't need the `else` block

File: src/components/CourseCatalog.jsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    return (
+      <>
+        <Button variant="primary" onClick={handleShow}>Take Course</Button>
+
+        <Modal show={show} onHide={handleClose}>
```
Comment (Russell-Pollari) [inline src/components/CourseCatalog.jsx:58]: UX question. Why put this in a Modal? Seems like the page is basically empty

File: src/components/Student.jsx [hunk 4]
```diff
@@ -38,7 +65,8 @@ const Student = () => {
                 <Card.Text>Program: {student.program}</Card.Text>
                 {
                   student.courses.map((course) => {
-                    <Badge Key={course.courseId} bg="secondary">
+                    <Badge key={course.code} bg="secondary"
+                      onClick={dropCourse(student.studentd, course.codes)}>
```
Comment (Russell-Pollari) [inline src/components/Student.jsx:69]: I think this should be more like `onClick={() => dropCourse(student.studentId, ...)}`

**Review by Russell-Pollari — CHANGES_REQUESTED** [review]

File: src/App.js (hunk 1 above)
Comment (Daniel-olaO) [reply src/App.js:57]: I added it to the nested route so that the navbar can show

File: src/components/CourseCatalog.jsx [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    method: 'GET',
+  })
+      .then((data) => data.json())
+      .catch((err)=>console.log(err));
+}
+function takeCourse(studentId, courseCode) {
+  const cookies = new Cookies();
+  const baseUrl = process.env.API_BASE_URL || 'http://localhost:8000';
+  return fetch(`${baseUrl}/api/students/
+  ${studentId}/takeCourse/${courseCode}`, {
```
Comment (Russell-Pollari) [inline src/components/CourseCatalog.jsx:22]:
```suggestion
  return fetch(`${baseUrl}/api/students/  ${studentId}/takeCourse/${courseCode}`, {
```

File: src/components/CourseCatalog.jsx [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+
+  const handleClose = () => setShow(false);
+  const handleShow = () => setShow(true);
+
+  useEffect(()=>{
+    getCourses()
+        .then((data) => {
+          setCourses(data);
+        },
+        )
```
Comment (Russell-Pollari) [inline src/components/CourseCatalog.jsx:44]: what's with the comma?
```suggestion
        })
```

File: src/components/Student.jsx [hunk 7]
```diff
@@ -1,25 +1,52 @@
...
+    },
+  })
+      .then((response) => response.json());
+}
+
+function dropCourse(studentId, courseCode) {
+  const cookies = new Cookies();
+  const baseUrl = process.env.API_BASE_URL || 'http://localhost:8000';
+  return fetch(`${baseUrl}/api/students/
+  ${studentId}/dropCourse/${courseCode}`, {
```
Comment (Russell-Pollari) [inline src/components/Student.jsx:25]:
```suggestion
  return fetch(`${baseUrl}/api/students/${studentId}/dropCourse/${courseCode}`, {
```

File: src/components/Student.jsx [hunk 8]
```diff
@@ -29,22 +56,28 @@ const Student = () => {
       <Container>
         <Row>
           {
```
Comment (Russell-Pollari) [inline src/components/Student.jsx:58]: what are these brackets for?

File: src/App.js (hunk 1 above)
Comment (Russell-Pollari) [reply src/App.js:57]: so do you need this route at all here?

**Review by Russell-Pollari — CHANGES_REQUESTED** [review]

**Review by Russell-Pollari — APPROVED** [review]

**Pull request merged**